Sources/OfflineEntitlements/PurchasedSK2Product.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  PurchasedSK2Product.swift
+//
+//  Created by Nacho Soto on 3/21/23.
+
+import Foundation
+import StoreKit
+
+/// Contains all information from a StoreKit 2 transaction necessary to create an ``EntitlementInfo``.
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+struct PurchasedSK2Product {
+
+    let productIdentifier: String
+    let subscription: CustomerInfoResponse.Subscription
+    let entitlement: CustomerInfoResponse.Entitlement
+
+}
+
+@available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+extension PurchasedSK2Product {
+
+    init(
+        from transaction: StoreKit.Transaction,
+        sandboxEnvironmentDetector: SandboxEnvironmentDetector = BundleSandboxEnvironmentDetector.default
+    ) {
+        let expiration = Date().addingTimeInterval(DispatchTimeInterval.days(1).seconds)
+
```

Comment:
why not use `transaction.expirationDate` here?

Reply:
Fixed.